src/Controllers/DatabaseController.js

```diff
@@ -616,13 +616,14 @@ DatabaseController.prototype.reduceInRelation = function(className, query, schem
   }
 
   const promises = Object.keys(query).map((key) => {
+    const t = schema.getExpectedType(className, key);
+    if (!t || t.type !== 'Relation') {
+      return Promise.resolve(query);
+    }
+    let queries = [{isNegation: false, relatedIds: []}];
```

Comment:
It is the trick that solves the problem. When it is null and not match this (https://github.com/parse-community/parse-server/pull/3924/files#diff-26b0f888e8a20f22d9728a26272f6cddR624), we need to pass relatedIds empty, otherwise it will be ignored. I changed little bit the code. Take a look if it is more clear now.